llvm/test/CodeGen/Hexagon/tail-dup-subreg-abort.ll

```diff
@@ -1,12 +1,18 @@
-; RUN: llc -march=hexagon -O2 -disable-cgp < %s
-; REQUIRES: asserts
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
+; RUN: llc -march=hexagon -O2 -disable-cgp < %s | FileCheck %s
 ;
 ; Tail duplication can ignore subregister information on PHI nodes, and as
 ; a result, generate COPY instructions between registers of different classes.
 ; This could lead to HexagonInstrInfo::copyPhysReg aborting on an unhandled
 ; src/dst combination.
 ;
 define i32 @foo(i32 %x, i64 %y) nounwind {
+; CHECK-LABEL: foo:
+; CHECK:       // %bb.0: // %entry
+; CHECK-NEXT:    {
+; CHECK-NEXT:     r1:0 = combine(r3,r2)
```

Comment:
This looks like too much was optimized out, that I doubt whatever MIR pattern was here didn't survive late enough

Reply:
Alright, so is there a way to see all the optimizations done

Reply:
-print-changed/-print-after-all?